src/matrix_cookbook/3_inverses.lean

```diff
@@ -86,13 +88,80 @@ end
 
 /-! ### The Searle Set of Identities -/
 
-lemma eq_161 : (1 + A⁻¹)⁻¹ = A⬝(A + 1)⁻¹ := sorry
-lemma eq_162 : (A + Bᵀ⬝B)⁻¹⬝B = A⁻¹⬝B⬝(1 + Bᵀ⬝A⁻¹⬝B)⁻¹:= sorry
-lemma eq_163 : (A⁻¹ + B⁻¹)⁻¹ = A⬝(A + B)⁻¹⬝B ∧ (A⁻¹ + B⁻¹)⁻¹ = B⬝(A + B)⁻¹⬝A := sorry
-lemma eq_164 : A - A⬝(A + B)⁻¹⬝A = B - B⬝(A + B)⁻¹⬝B := sorry
-lemma eq_165 : A⁻¹ + B⁻¹ = A⁻¹⬝(A + B)⁻¹⬝B⁻¹ := sorry
-lemma eq_166 : (1 + A⬝B)⁻¹ = 1 - A⬝(1 + B⬝A)⁻¹⬝B := sorry
-lemma eq_167 : (1 + A⬝B)⁻¹⬝A = A⬝(1 + B⬝A)⁻¹ := sorry
+lemma eq_161 {hA: is_unit A.det} {hAaddOne: is_unit (A+1).det} :  (1 + A⁻¹)⁻¹ = A⬝(A + 1)⁻¹ :=
+begin
+  rw inv_eq_right_inv, 
+  rw [matrix.add_mul, matrix.one_mul, nonsing_inv_mul_cancel_left],
+  nth_rewrite 1 ← matrix.one_mul (A + 1)⁻¹,   
+  rw [← matrix.add_mul, mul_nonsing_inv],
+  assumption',
+end
+lemma eq_162 {B: matrix n m ℂ} {hAB: is_unit (1 + Bᵀ ⬝ A⁻¹ ⬝ B).det} : 
+  (A + B⬝Bᵀ)⁻¹⬝B = A⁻¹⬝B⬝(1 + Bᵀ⬝A⁻¹⬝B)⁻¹:= 
+begin
+  rw [eq_159, matrix.sub_mul, sub_eq_iff_eq_add],
+  repeat {rw matrix.mul_assoc (A⁻¹⬝B), },
+  rw ← matrix.mul_add (A⁻¹⬝B),
+  nth_rewrite 0 ← matrix.mul_one (1 + Bᵀ ⬝ A⁻¹ ⬝ B)⁻¹,
+  repeat {rw matrix.mul_assoc (1 + Bᵀ ⬝ A⁻¹ ⬝ B)⁻¹, },
+  rwa [← matrix.mul_add (1 + Bᵀ ⬝ A⁻¹ ⬝ B)⁻¹, nonsing_inv_mul, matrix.mul_one],
+end
+lemma eq_163 {hA: is_unit A.det} {hB: is_unit B.det} {hAB: is_unit (A + B).det}: 
+  (A⁻¹ + B⁻¹)⁻¹ = A⬝(A + B)⁻¹⬝B ∧ (A⁻¹ + B⁻¹)⁻¹ = B⬝(A + B)⁻¹⬝A :=
+begin
+  split,
+  work_on_goal 2 {swap_var [A B], rw add_comm, rw add_comm B,  },
+  all_goals 
+  { rw inv_eq_right_inv, 
+    rw [matrix.add_mul, matrix.mul_assoc, nonsing_inv_mul_cancel_left],
+    apply_fun (λ x, B⬝x), dsimp, 
+    rw [matrix.mul_add, mul_nonsing_inv_cancel_left, ← matrix.add_mul, add_comm,
+    mul_nonsing_inv_cancel_left,matrix.mul_one],},
+  assumption',
+  exact left_mul_inj_of_is_unit_det hB,
+  rw add_comm, exact hAB,
+  exact left_mul_inj_of_is_unit_det hA,
+end
+
+lemma eq_164_one_side (A B : matrix n n ℂ) (hA: is_unit A.det) (hB: is_unit B.det): 
+  A - A⬝(A + B)⁻¹⬝A = (B⁻¹ + A⁻¹)⁻¹ := 
+begin
+  haveI invB := invertible_of_is_unit_det B hB,
+  nth_rewrite 0 ← matrix.mul_one B,
+  rw [eq_159, matrix.one_mul, matrix.mul_one, matrix.mul_sub, mul_nonsing_inv A hA, 
+    matrix.sub_mul, matrix.one_mul, sub_sub_cancel,  matrix.mul_assoc _ _ A, 
+    nonsing_inv_mul_cancel_right A _ hA, matrix.mul_assoc, mul_nonsing_inv_cancel_left A _ hA],
+  nth_rewrite 0 ← inv_inv_of_invertible B,
+  rw [← matrix.mul_inv_rev, matrix.add_mul, matrix.one_mul, mul_nonsing_inv_cancel_right B _ hB],
+end
+
+lemma eq_164 {hA: is_unit A.det} {hB: is_unit B.det}: 
+  A - A⬝(A + B)⁻¹⬝A = B - B⬝(A + B)⁻¹⬝B :=
+begin
+  rw [eq_164_one_side A B hA hB, add_comm A, eq_164_one_side B A hB hA, add_comm],
+end
+
+lemma eq_165 {hA: is_unit A.det} {hB: is_unit B.det} : A⁻¹ + B⁻¹ = A⁻¹⬝(A + B)⬝B⁻¹ := 
```

Comment:
I made https://github.com/leanprover-community/mathlib4/pull/13215 to capture the more general case of this one upstream